core/src/services/fs/backend.rs

```diff
@@ -312,7 +320,19 @@ impl Access for FsBackend {
         let f = open_options
             .open(tmp_path.as_ref().unwrap_or(&target_path))
             .await
-            .map_err(new_std_io_error)?;
+            .map_err(|e| {
+                match e.kind() {
+                    std::io::ErrorKind::AlreadyExists => {
+                        // Map io AlreadyExists to opendal ConditionNotMatch
+                        Error::new(
+                            ErrorKind::ConditionNotMatch,
+                            ErrorKind::ConditionNotMatch.to_string(),
```

Comment:
How about generating a better error message for users? For example, "The file already exists in the filesystem."